### PR TITLE
[SCRIPT_ONLY] Removed JBOSS_HOME from narayana.sh

### PR DESCRIPTION
**Pull request by jmfinelli**
CORE LRA AS_TESTS QA_JTA

!TOMCAT !RTS !JACOCO !XTS !QA_JTS_OPENJDKORB !PERF !DB_TESTS !mysql !db2 !postgres !oracle

This PR is for improving `narayana.sh`, especially when running it locally. Please note that I set up the `main` branch to be the default of the repo [jboss-as](https://github.com/jbosstm/jboss-as)

### CODE REVIEW

File: scripts/hudson/narayana.sh [hunk 1]
```diff
@@ -408,15 +391,20 @@ function build_as {
...
+    [ "$_jdk" -lt 17 ] && export MAVEN_OPTS="-XX:MaxMetaspaceSize=512m -XX:+UseConcMarkSweepGC $MAVEN_OPTS"
+    [ "$_jdk" -ge 17 ] && export MAVEN_OPTS="-XX:MaxMetaspaceSize=512m $MAVEN_OPTS"
+    JAVA_OPTS="-Xms1303m -Xmx1303m -XX:MaxMetaspaceSize=512m $JAVA_OPTS" ./build.sh clean install -B -DskipTests -Dts.smoke=false $IPV6_OPTS -Dversion.org.jboss.narayana=${NARAYANA_CURRENT_VERSION} "$@"
+    [ $? -eq 0 ] || fatal "AS build failed"
+
+  fi
+
   echo "AS version is ${WILDFLY_VERSION_FROM_JBOSS_AS}"
-  JBOSS_HOME=${WILDFLY_CLONED_REPO}/build/target/wildfly-${WILDFLY_VERSION_FROM_JBOSS_AS}
+  JBOSS_HOME=${WILDFLY_CLONED_REPO}/dist/target/wildfly-${WILDFLY_VERSION_FROM_JBOSS_AS}
```
Comment (marcosgopen) [inline scripts/hudson/narayana.sh:407]: Thanks @jmfinelli  for your PR! Is the Wildfly build folder name changed?

Comment (jmfinelli) [reply scripts/hudson/narayana.sh:407]: Although both folders are provided with a `build` of WildFly, `dist` worked better in some tests.

File: scripts/hudson/narayana.sh [hunk 2]
```diff
@@ -349,25 +349,8 @@ function clone_as {
...
+    echo "Using existing checkout of WildFly. If a fresh build should be used, delete the folder ${WORKSPACE}/jboss-as"
     cd jboss-as
-
-    git remote | grep upstream
-    if [ $? -ne 0 ]; then
-      git remote add upstream https://github.com/wildfly/wildfly.git
-    fi
-    #Abort any partially complete rebase
-    git rebase --abort
-    git checkout 5_BRANCH
```
Comment (marcosgopen) [inline scripts/hudson/narayana.sh:361]: I am not sure why it used to checkout the 5_BRANCH, but I think it is related to the Narayana version. I suppose now we should have a 6_BRANCH and set it as default.  WDYT @jmfinelli ?

Comment (jmfinelli) [reply scripts/hudson/narayana.sh:361]: I think that the `main` branch should be used (for the current version of Narayana) and a maintenance branch (e.g. `5_BRANCH`) should be created when needed.

**Review by marcosgopen — APPROVED** [review]
Thanks @jmfinelli for answering to my comments. Your PR LGTM!

**Comment by marcosgopen** [comment]
AS_TESTS failure is not related to this PR, I will have a look at it BTW.

**Pull request merged**